src/main/java/com/google/sps/managers/DatastoreManager.java

```diff
@@ -0,0 +1,327 @@
+    gameEntity.setProperty("stages", game.getStages());
+    gameEntity.setProperty("numTimesPlayed", game.getNumTimesPlayed());
+    gameEntity.setProperty("numTimesFinished", game.getNumTimesFinished());
+    gameEntity.setProperty("numStarVotes", game.getNumStarVotes());
+    gameEntity.setProperty("totalStars", game.getTotalStars());
+    gameEntity.setProperty("numDifficultyVotes", game.getNumDifficultyVotes());
+    gameEntity.setProperty("totalDifficulty", game.getTotalDifficulty());
+    datastore.put(gameEntity);
+  }
+  
+  /**
+    * Retrieves a single game entity from the datastore.
+    * @param gameID the unique code used to identify this specific game.
+    * @return a Game object with the properties specified in the Builder.
+  */
+  public Game retrieveGame(String gameID) {
+    Key gameEntityKey = KeyFactory.createKey("Game", gameID);
+    Entity gameEntity;
+    try {
+      gameEntity = datastore.get(gameEntityKey);
+    }
+    catch(Exception e) {
+      return null;
+    }
+
+    String gameName = (String) gameEntity.getProperty("gameName");
+    String gameDescription = (String) gameEntity.getProperty("gameDescription");
+    String gameCreator = (String) gameEntity.getProperty("gameCreator");
+    ArrayList<String> stages = (ArrayList<String>)gameEntity.getProperty("stages");
+    int numTimesPlayed = (int)gameEntity.getProperty("numTimesPlayed");
+    int numTimesFinished = (int)gameEntity.getProperty("numTimesFinished");
+    int numStarVotes = (int)gameEntity.getProperty("numStarVotes");
+    int totalStars = (int)gameEntity.getProperty("totalStars");
+    int numDifficultyVotes = (int)gameEntity.getProperty("numDifficultyVotes");
+    int totalDifficulty = (int)gameEntity.getProperty("totalDifficulty");
+  
+    Game.Builder game = new Game.Builder(gameID, gameName).setGameCreator(gameCreator);
+    game.setGameDescription(gameDescription).setNumTimesPlayed(numTimesPlayed).setNumTimesFinished(numTimesFinished);
+    game.setNumStarVotes(numStarVotes).setTotalStars(totalStars).setNumDifficultyVotes(numDifficultyVotes).setTotalDifficulty(totalDifficulty);       
+    return game.build();
+  }
+
+  /**
+    * Creates the static data of a single stage in datastore as an entity
+    * @param game an Stage variable representing a single instance of a stage.
+  */
+  public void createStage(Stage stage) {
+    ArrayList<String> hints = new ArrayList<String>();
+    Entity stageEntity = new Entity("Stage", stage.getStageID());
+    stageEntity.setProperty("stageNumber", stage.getStageNumber());
+    stageEntity.setProperty("key", stage.getKey());
+    stageEntity.setProperty("startingHint", stage.getStartingHint());
+    stageEntity.setProperty("latitude", stage.getStartingLocation().getLatitude());
+    stageEntity.setProperty("longitude", stage.getStartingLocation().getLongitude());
+    for (Hint hint : stage.getHints()) {
+      createOrReplaceHint(hint);
+    }
+    for (Hint hint : stage.getHints()) {
```

Comment:
Combine this loop with the one above since they are looping over the same thing.